iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -143,19 +145,30 @@ public void commitJob(JobContext originalContext) throws IOException {
     LOG.info("Committing job has started for table: {}, using location: {}", table,
         generateJobLocation(conf, jobContext.getJobID()));
 
-    FileIO io = HiveIcebergStorageHandler.io(jobContext.getJobConf());
-    List<DataFile> dataFiles = dataFiles(jobContext, io, true);
-
-    if (dataFiles.size() > 0) {
+    List<DataFile> dataFiles = dataFiles(jobContext, table.io(), true);
+
+    boolean isOverwrite = conf.getBoolean(InputFormatConfig.IS_OVERWRITE, false);
+    if (isOverwrite) {
+      // Replacing affected partitions (or whole table) since it's an insert overwrite
+      // We do the replacement even for 0 data files (i.e. empty source)
+      ReplacePartitions overwrite = table.newReplacePartitions();
+      dataFiles.forEach(overwrite::addFile);
+      overwrite.commit();
```

Comment:
Is there an easy way to check if we change anything?

Reply:
For logging/debugging? Not sure, I don't think we can check it without doing FS calls

Reply:
Actually, there is `ReplacePartitions validateAppendOnly()`, but I'm not sure yet what it does. Let me take a look

Reply:
More like not committing if not necessary...
But with the funny logic around partitions this sounds hard to check anyway

Reply:
Never mind, it just makes sure that an exception is thrown if we're overwriting (deleting) any existing data

Reply:
For unpartitioned table, we still want to commit because the semantics is that the empty result set should clear the table, that's why I kept the commit even for 0 data files